src/main/java/org/opensearch/neuralsearch/search/HybridQueryTopDocs.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.search;
+
+import java.util.Arrays;
+
+import lombok.Getter;
+import lombok.ToString;
+
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TotalHits;
+
+/**
+ * Class stores collection of TodDocs for each sub query from hybrid query
+ */
+@ToString(includeFieldNames = true)
+public class HybridQueryTopDocs extends TopDocs {
```

Comment:
Is HybridQueryTopDocs the right name? This class is basically a TopDocs implementation that stores multiple TopDocs - so maybe CompondTopDocs would be better?

Reply:
I dont think the name of this class needs to map to the query api. It should be descriptive of its functionality. I think CompoundTopDocs makes sense.

Reply:
I actually put it as CompoundTopDocs in design initially, but then changed to hybrid to make it consistent. If at least one more person came to same idea I'll change it to CompoundTopDocs, will push commit soon